packages/libs/react-ui/src/components/Typography/Text/Text.tsx

```diff
@@ -14,70 +15,89 @@ import {
   monospaceSmallRegular,
   monospaceSmallestBold,
   monospaceSmallestRegular,
+  uiBaseBold,
+  uiBaseRegular,
+  uiSmallBold,
+  uiSmallRegular,
+  uiSmallestBold,
+  uiSmallestRegular,
 } from '../../../styles';
 import { colorVariants, transformVariants } from '../typography.css';
 
-// eslint-disable-next-line @kadena-dev/typedef-var
-export const TEXT_ELEMENTS = ['p', 'span', 'code'] as const;
-export type TextElementType = (typeof TEXT_ELEMENTS)[number];
-
-type TextVariant = 'small' | 'smallest' | 'base';
-function getFontClass(
-  variant: TextVariant,
-  isBold: boolean,
-  type: TextElementType,
-): string {
-  if (type === 'code' && variant === 'smallest') {
-    return isBold ? monospaceSmallestBold : monospaceSmallestRegular;
-  }
-  if (type === 'code' && variant === 'small') {
-    return isBold ? monospaceSmallBold : monospaceSmallRegular;
-  }
-
-  if (type === 'code' && variant === 'base') {
-    return isBold ? monospaceBaseBold : monospaceBaseRegular;
-  }
-
-  if (variant === 'smallest') {
-    return isBold ? bodySmallestBold : bodySmallestRegular;
-  }
-  if (variant === 'small') {
-    return isBold ? bodySmallBold : bodySmallRegular;
-  }
-
-  return isBold ? bodyBaseBold : bodyBaseRegular;
-}
-
-export interface ITextProps extends ComponentPropsWithRef<'p'> {
-  as?: TextElementType;
-  variant?: TextVariant;
-  bold?: boolean;
+export interface ITextProps
+  extends HTMLAttributes<HTMLSpanElement | HTMLParagraphElement> {
+  as?: 'p' | 'span' | 'code';
+  children: string | ReactNode;
+  className?: string;
   color?: keyof typeof colorVariants;
+  bold?: boolean;
+  size?: 'small' | 'smallest' | 'base';
   transform?: keyof typeof transformVariants;
-  children: React.ReactNode;
+  variant?: 'body' | 'code' | 'ui';
+  ariaLabel?: HTMLAttributes<HTMLSpanElement>['aria-label'];
+  id?: HTMLAttributes<HTMLSpanElement>['id'];
 }
 
-export const Text: FC<ITextProps> = ({
-  as = 'span',
-  variant = 'base',
-  bold = false,
-  color = 'default',
-  transform = 'none',
+const fontMap = {
```

Comment:
nit: I believe you can pull this out of the component itself

Reply:
I'll create a constants file for this one. This will probably happen at other components as well, then I can keep it consistent.